src/mesh/amr_loadbalance.cpp

```diff
@@ -705,13 +705,27 @@ void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput
             if (ranklist[on + ll] != Globals::my_rank) continue;
             // fine to coarse on the same MPI rank (different AMR level) - restriction
             auto pob = FindMeshBlock(on + ll);
+
+            // allocte sparse variables that were allocated on old block
+            for (auto var : pob->meshblock_data.Get()->GetCellVariableVector()) {
+              if (var->IsSparse() && var->IsAllocated()) {
+                new_block_list[n - nbs]->AllocateSparse(var->label());
+              }
+            }
```

Comment:
we may not actually always need the sparse field on every child, but it's fine for now.  if there's nothing there, it'll deallocate after a few cycles anyway, right?

Reply:
I was actually thinking of the case where the sparse field is only in, say, one corner of the parent block.  So some of the child blocks would have all zeros.